### DISCUSSION TITLE
Data Table Component : how to hide certain column based on some condition

### DISCUSSION THREAD

**Discussion by shrix1** [shadcn-ui/ui#771] [Q&A] [answered]
### For Example : i like to show the view video/ppt column in table only for status : past 

![image](https://github.com/shadcn/ui/assets/92677078/26df0f2b-3c71-4707-a680-bee3c878003e)

# can anyone help on this

**Comment by noxify** [comment, marked as answer]
Hi @shrix1 ,

i don't know how your filter ( Upcoming, Past, Canceled ) is implemented but let's assume you're using an url parameter to handle the status filter ( e.g. `http://localhost/list?status=past` ), I see two possible ways to solve this.

Based on the url parameter `status` you can 

* use a different column list which excludes the `View Video/PPT` column
* use the column visibility ( which is documented [here](https://ui.shadcn.com/docs/components/data-table#visibility) and [here](https://tanstack.com/table/v8/docs/api/features/column-visibility) ) to hide the column.
   ```js
     const [columnVisibility, setColumnVisibility] = React.useState({
        viewVideoPpt: false, 
     });
   ```

Personally, I would prefer option 2.

**Reply by shrix1** [reply]
Thanks @noxify
